src/borg/archiver/debug_cmd.py

```diff
@@ -265,6 +265,71 @@ def do_debug_id_hash(self, args, repository, manifest):
         print(id.hex())
         return EXIT_SUCCESS
 
+    @with_repository(compatibility=Manifest.NO_OPERATION_CHECK)
+    def do_debug_parse_obj(self, args, repository, manifest):
+        """parse object file into meta dict and data (decrypting, decompressing)"""
+
+        # get the object from id
+        hex_id = args.id
+        try:
+            id = unhexlify(hex_id)
+            if len(id) != 32:  # 256bit
+                raise ValueError("id must be 256bits or 64 hex digits")
+        except ValueError as err:
+            print("object id %s is invalid [%s]." % (hex_id, str(err)))
+            return EXIT_ERROR
+        try:
+            cdata = repository.get(id)
+        except Repository.ObjectNotFound:
+            print("object %s not found." % hex_id)
+            return EXIT_ERROR
+
+        output_metadata_path: str = args.json_path
+        output_data_path: str = args.data_path
+
+        key = manifest.key
+        repo_objs = RepoObj(key)
+
+        try:
+            meta, data = repo_objs.parse(obj_id=id, cdata=cdata, decompress=True)
+        except Exception as e:
+            print(f"encountered exception {e.__str__}")
```

Comment:
would just `{e}` also work? if not, please use `str(e)`.

Reply:
Yes. Also should I use `repr(e)` since it may provide more information about the error? ([link](https://stackoverflow.com/questions/37684153/convert-exception-error-to-string)

Reply:
I am using [traceback.print_exc()](https://docs.python.org/3/library/traceback.html) because if this is a debug command, the user may be interested in knowing where the error occurs. {e} also works but it just provides the classname of the error.